app/src/main/java/xyz/whereuat/whereuat/AutoShareStar.java

```diff
@@ -35,26 +50,61 @@ public AutoShareStar(Context context, AttributeSet attrs) {
         mPath = new Path();
     }
 
-    public void toggleFilled() {
-        mIsFilled = !mIsFilled;
+    public void toggleAutoShare() {
+        mIsAutoShared = !mIsAutoShared;
         postInvalidate();
         requestLayout();
     }
 
+    /*
+        When clicked, the contact's autoshare status will be updated and this star will be redrawn
+        along with the star on the other side of the contact's ContactCard.
+     */
     @Override
     public boolean onTouchEvent(final MotionEvent event) {
-        if(event.getAction() == MotionEvent.ACTION_UP && mIsClickable){
-            this.toggleFilled();
+        if(event.getAction() == MotionEvent.ACTION_UP && mIsClickable) {
+            ContentValues updated_val = new ContentValues();
+            updated_val.put(ContactEntry.COLUMN_AUTOSHARE, !mIsAutoShared);
+
+            String where = String.format("%s=?", ContactEntry._ID);
+            final ContactCard contact_card = (ContactCard) this.getParent().getParent().getParent();
+            String contact_id = contact_card.getTag().toString();
+            String[] where_args = new String[] {contact_id};
+            UpdateCommand update = new UpdateCommand(mContext, ContactEntry.TABLE_NAME, updated_val,
+                    where, where_args);
```

Comment:
Could make an update() method in the Contact class and cut down on the table name arg